test/Test/MockApi.hs

```diff
@@ -126,15 +146,23 @@ newtype MockApiC m a = MockApiC
 instance (Algebra sig m, Has (Lift IO) sig m) => Algebra (MockApi :+: sig) (MockApiC m) where
   alg hdl sig ctx = MockApiC $ case sig of
     L (MockApiOnce req resp) -> do
-      let expectation = ApiExpectation Once req (ApiResult (Right resp))
+      let expectation = ApiExpectation Once ExpectingExactRequest req (ApiResult (Right resp))
```

Comment:
This is starting to look repetitive and we could probably pull it out into a helper since 2/3 lines are the same in every case.